docs/tutorials/utils/simulateUserOp.md

```diff
@@ -0,0 +1,42 @@
+---
+sidebar_label: "Simulate User Op"
+sidebar_position: 3
+title: "Simulate a User Op with buildUserOp()"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+### Overview
+
+This tutorial provides insights into how to simulate a user operation using the buildUserOp() method.
+
+
+```typescript
+const account = privateKeyToAccount("");
+
+const signer = createWalletClient({
+    account,
+    chain: polygonAmoy,
+    transport: http(),
+})
+
+const smartAccount = await createSmartAccountClient({
+    signer,
+    bundlerUrl, // from Biconomy dashboard
+    paymasterUrl, // from Biconomy dashboard
+});
+
+const encodedCall = encodeFunctionData({
+    abi: parseAbi(["function safeMint(address _to)"]),
+    functionName: "safeMint",
+    args: [recipient]
+})
+const transaction = {
+    to: nftAddress, 
+    data: encodedCall
+}
+
+// will throw an error if something is wrong with the transaction
+await smartAccount.buildUserOp([transaction]);
```

Comment:
we will need to pass simulationType here right?

Reply:
No, actually, simulationType will be removed from the SDK in the next version, turns out infra simulates both validation and execution no matter what.

Reply:
I see then lets remove it from here as well https://docs.biconomy.io/Account/methods#sendtransaction- here we have added simulationType

Reply:
https://docs.biconomy.io/Account/methods#senduserop-

Reply:
@arcticfloyd1984 should we change this as well https://docs.biconomy.io/Bundler/bundlermethods#senduserop ?

Reply:
The initial idea of `validation` and `validation_and_execution` was done way before the gas estimations were worked on. 
Back then validation would just call the `simuateHandleOp` and it would just run the validation checks and `validation_and_execution` uses Tenderly and Alchemy Simulation APIs to capture the events and check for execution (call data execution) reverts.

Once we improved gas estimation using state overrides, we are now able to check everything in the default flow (which from the previous context is validation flow) so going forward we can remove the simulation_type altogether from the SDK and Bundler APIs.
I will not remove it from the Bundler code as it may break things, but we can remove it from SDK and not reflect it in docs anymore. The Bundler API can accept a request without this param also so not sending it runs into the default flow. So I think we can remove from the SDK.


Also as you guys might have noticed on not-so-famous chains the state overrides don't work and neither did the Simulation APIs so there would be chains where only the validation phase will be checked.

Reply:
The reason why I wouldn't remove it from SDK right now is because it would cause breaking changes which will require us to bump the version of SDK to v5. I propose instead to mark it as deprecated and remove it in the next major version of the SDK which will be Nexus so that we stay consistent with versioning.

Reply:
@himanshugarg06 we can mark it as deprecated in the dos as well or simply remove it.